### ISSUE TITLE
pretrained model 'darknet53.conv.74'

### ISSUE THREAD

**Issue by jcdubron** [AlexeyAB/darknet#5433]
I find that [darknet53.conv.74](https://pjreddie.com/media/files/darknet53.conv.74) is not the same as [darknet53.weights](https://pjreddie.com/media/files/darknet53.weights) and [darknet53_448.weights](https://pjreddie.com/media/files/darknet53_448.weights). Is `darknet53.conv.74` trained in ImageNet with different settings or other random seed?

**Comment by WongKinYiu** [comment]
`darknet53.weights` train imagenet from scratch with image size 256x256.
`darknet53.conv.74` is the weights of first 74 layers of `darknet53.weights`.
`darknet53_448.weights` is fine-tune with image size 448x448 on `darknet53.weights`.

**Comment by jcdubron** [comment]
But I find the values in `darknet53.conv.74` is not the same as the first 74 layers of `darknet53.weights`. The first 5 weights in `darknet53.weights` is [-0.11338928  1.1283484   1.2455083   1.9338647  -4.148237] and the first 5 weights in `darknet53.conv.74` is [-4.1455936  -0.99273944 -2.060937    1.2925771  -0.89972943].

**Comment by WongKinYiu** [comment]
what is the first 5 weights of `darknet53_448.weights`?

**Comment by jcdubron** [comment]
The first 5 weights of `darknet53_448.weights` are [-0.10601279  1.1518075   1.2749087   1.9374714  -4.1375694].

**Comment by WongKinYiu** [comment]
well, it becomes a mystery.
![image](https://user-images.githubusercontent.com/12152972/80797031-4ded9e80-8bd3-11ea-8c60-b933dfa1399a.png)
maybe you are correct.

**Comment by AlexeyAB** [comment]
Just to know the first 20 bytes from .weights-file are not the weights: https://github.com/AlexeyAB/darknet/blob/f14054ec2b49440ad488c3e28612e7a76780bc5f/src/parser.c#L1763-L1767

**Comment by jcdubron** [comment]
@AlexeyAB That's right. I ignore the first 20 bytes but they are still not the same.

**Comment by AlexeyAB** [comment]
This is a question for Joe)

**Comment by WongKinYiu** [comment]
darknet53: 102400000 = 128x800000
darknet53.conv.74: partial will clean the information
darknet53_448: 12800000 = 128x800000 (darknet53 pre-train)+ 128x200000 (448 fine-tune)

**Comment by nguyenanhtuan1008** [comment]
I have a question. What is the relationship between 53 and 74? How I get the number with darknet19_448?

**Comment by WongKinYiu** [comment]
53 means there are totally 53 convolutional layers in darknet53.
74 means 53-1 (convolution for classify) + 23-1 (last shortcut layer has no weights) = totally 74 layers.

for darknet19, you can get number by calculate 19-1 (convolution for classify) + number of maxpooling layers.
![image](https://user-images.githubusercontent.com/12152972/95525371-7d1bfd80-0a06-11eb-9e50-80e06010a547.png)

**Comment by nguyenanhtuan1008** [comment]
> 53 means there are totally 53 convolutional layers in darknet53.
> 74 means 53-1 (convolution for classify) + 23-1 (last shortcut layer has no weights) = totally 74 layers.
> 
> for darknet19, you can get number by calculate 19-1 (convolution for classify) + number of maxpooling layers.
> ![image](https://user-images.githubusercontent.com/12152972/95525371-7d1bfd80-0a06-11eb-9e50-80e06010a547.png)

Thank you so much for your answer. So, in this case, darknet19 will be 19-1 + 5-1 = 22, right?

**Comment by WongKinYiu** [comment]
23

**Comment by nguyenanhtuan1008** [comment]
> 23
Thank you!
So what is the layer name of ”last shortcut layer has no weights” in darknet53
![image](https://user-images.githubusercontent.com/36155295/95658393-0d685880-0b55-11eb-9d3a-81f04dcf9339.png)

**Comment by WongKinYiu** [comment]
the layer just before avgpol.
![image](https://user-images.githubusercontent.com/12152972/95669924-27875280-0bb8-11eb-8722-f9e7af96f0fb.png)

**Comment by nguyenanhtuan1008** [comment]
@WongKinYiu Thank you for your answer!
